### PR TITLE
Add GitHub remote config plugin

### PR DESCRIPTION
**Pull request by ParkerM**
Was working on a remote config plugin but having some trouble getting it to execute on boot. The function is called on boot (line 105) but api_key doesn't get hooked in so it fails. The api_key hook only seems to work when the function is called via the command hook. If you have any ideas on how to make the api_key hook work on init/reload please share.

The goal was to have a private repo storing our config.json, so the only actual private key you would need to run the bot locally is your github token (and access to the private config repo).

Edit: Too used to GitLab... By WIP: I mean please don't merge, Just wanted to bring this to your attention.

### CODE REVIEW

**Comment by ParkerM** [comment]
This is working now by adding a githubconfig.json file in the base directory (and makes more sense I suppose rather than reparsing the entire config just to save the githubconfig keys). I amended the commit so the stuff I said above might not make sense.

**Comment by imnotjames** [comment]
I'm kinda iffy about the remote repo for config.  It becomes hard to manage and also means loading stuff at random onto my local server in weird ways.  If we were constantly changing the config, maybe, but we aren't.

In other words: What problem is this solving

**Comment by ParkerM** [comment]
The idea was to give other members of the config repo the ability to temporarily rehost helpdesk if the server goes down without having to reconfigure everything locally. I've used a similar setup with spring cloud config for collaborative projects and it's a big help having everything centrally stored somewhere, but in those cases the keys were tied to a shared service account, so I understand the hesitance in our case (since most of the keys are tied to personal accounts).

Either way just thought it was a neat idea.

File: plugins/githubconfig.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+CONFIG_PATH_TEMPLATE = REPO_API_PATH + '/{owner}/{repo}/contents/{config_path}'
+LAST_COMMIT_PATH_TEMPLATE = REPO_API_PATH + '/{owner}/{repo}/git/refs/heads/{branch}'
+
+@hook.command('refreshconfig')
+def githubconfig(inp):
+    global GITHUB_CONFIG
+    required_keys = ['owner', 'repo', 'private_token']
+
+    if GITHUB_CONFIG is None or None in required_keys:
```
Comment (ParkerM) [inline plugins/githubconfig.py]: Just realized `None in required_keys` will always be false.

**Comment by imnotjames** [comment]
It's a neat idea but some of these are tied to accounts with my CC information attached which makes me hesitant.

Are you certain that rewriting the config causes a reload of the bot from config?  I never seem to remember that being the case.

**Comment by ParkerM** [comment]
Yeah, see [here](https://github.com/TeamPeggle/ppp-helpdesk/blob/ppp-helpdesk/bot.py#L47) (and [here](https://github.com/TeamPeggle/ppp-helpdesk/blob/ppp-helpdesk/core/config.py#L48)). Feel free to close though because the account/CC thing makes it sound like a bad idea.

**Pull request closed without merging**